src/transformers/trainer.py

```diff
@@ -260,7 +268,7 @@ def get_train_dataloader(self) -> DataLoader:
         )
 
     def _get_eval_sampler(self, eval_dataset: Dataset) -> Optional[torch.utils.data.sampler.Sampler]:
-        if isinstance(eval_dataset, torch.utils.data.IterableDataset):
+        if not isinstance(eval_dataset, collections.abc.Sized):
```

Comment:
This case should use the same defensive test as in init: if the user provides a non-sized evaluation/test set, there should be a ValueError here as the rest won't work.

Reply:
Done